code/modules/power/heavy_emitter.dm

```diff
@@ -0,0 +1,261 @@
+	var/obj/machinery/power/heavy_emitter/cannon/linked_cannon
+	///Is this currently firing?
+	var/firing = FALSE
+	///Cooldown
+	var/timer = 0
+	///Cooldown threshold
+	var/max_timer = 10
+
+/obj/machinery/power/heavy_emitter/centre/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj,src)
+
+/obj/machinery/power/heavy_emitter/centre/examine(mob/user)
+	. = ..()
+	if(firing)
+		. += "It is currently firing"
+	else
+		. += "It is currently turned off"
+
+	if(!is_fully_constructed)
+		. += "Insert a pyroclastic anomaly core to fuel the core!"
+
+/obj/machinery/power/heavy_emitter/centre/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W,/obj/item/assembly/signaler/anomaly/pyro) && !is_fully_constructed)
+		is_fully_constructed = TRUE
+		qdel(W)
+
+/obj/machinery/power/heavy_emitter/centre/on_set_is_operational(old_value)
+	. = ..()
+	if(is_operational)
+		turn_on()
+	else
+		turn_off()
+
+/obj/machinery/power/heavy_emitter/centre/check_part_connectivity()
+	. = ..()
+	if(!anchored)
+		return FALSE
+	for(var/obj/machinery/power/heavy_emitter/object in orange(1,src))
+		if(. == FALSE)
+			break
+		if(!object.anchored)
+			. = FALSE
+		if(istype(object,/obj/machinery/power/heavy_emitter/arm))
+			var/dir = get_dir(src,object)
+			if(dir in GLOB.cardinals)
+				. =  FALSE
+			switch(dir)
+				if(SOUTHEAST)
+					if(object.dir != SOUTH)
+						. = FALSE
+				if(SOUTHWEST)
+					if(object.dir != WEST)
+						. =  FALSE
+				if(NORTHEAST)
+					if(object.dir != EAST)
+						. =  FALSE
+				if(NORTHWEST)
+					if(object.dir != NORTH)
+						. =  FALSE
+			continue
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/power/heavy_emitter/vent))
+			//we dont want an object to appear twice in here
+			vents |= object
+
+		if(istype(object,/obj/machinery/power/heavy_emitter/interface))
+			if(linked_interface && linked_interface != object)
+				. =  FALSE
+			linked_interface = object
+
+		if(istype(object,/obj/machinery/power/heavy_emitter/cannon))
+			if(linked_cannon && linked_cannon != object)
+				. =  FALSE
+			linked_cannon = object
+
+/obj/machinery/power/heavy_emitter/centre/should_have_node()
+	return anchored
+
+/obj/machinery/power/heavy_emitter/centre/connect_to_network()
+	if(!anchored)
+		return FALSE
+	. = ..()
+
+/obj/machinery/power/heavy_emitter/centre/turn_on()
+	if(!is_fully_constructed)
+		return linked_interface.turn_off()
+	use_power = IDLE_POWER_USE
+	firing = TRUE
+	icon_state = "centre"
+
+/obj/machinery/power/heavy_emitter/centre/turn_off()
+	firing = FALSE
+	icon_state = "centre_off"
+
+/obj/machinery/power/heavy_emitter/centre/process(delta_time)
+	if(!firing || machine_stat & BROKEN || surplus() < active_power_usage)
+		return
+	if(!check_part_connectivity())
+		return linked_interface.turn_off()
+
+	timer += delta_time
+
+	add_load(idle_power_usage)
+	if(timer >= max_timer)
+		timer = 0
+		add_load(active_power_usage)
+		radiation_pulse(src,500,can_contaminate=FALSE)
```

Comment:
How does this work with dropoff?

Reply:
it doesnt release enough radiation to sustain itself meaning it is a general hazard and always required an external power source, it also doesnt contaminate preventing further spreading

Reply:
is it a significant threat 2 tiles away from the center? do you notice it when using the machine?

Reply:
yes it is a threat, since machinery doesnt block rads